heuristic/heuristic_calculations.py

```diff
@@ -1,322 +1,211 @@
 
     for key in off_shift_periods:
-        w[key] = max(off_shift_periods[key], key=itemgetter(1))
+        w[key] = max(off_shift_periods[key],key=itemgetter(1))
 
 
-def calculate_negative_deviation_from_demand(data, y):
 
-    employees_with_competencies = data["staff"]["employees_with_competencies"]
+def calculate_negative_deviation_from_demand(model, y):
     delta = {}
-
-    for c in data["competencies"]:
-        for i in data["time"]["days"]:
-            # todo: increase the readability of this set
-            for t in data["time"]["periods"][2][i]:
-                delta[c, t] = max(
-                    0,
-                    data["demand"]["ideal"][c, t]
-                    - sum(y[c, e, t] for e in employees_with_competencies[c]),
-                )
+    for c in model.competencies:
+        for i in model.days:
+            for t in model.time_periods_in_day[i]:
+                delta[c,t] = max(0, model.demand["ideal"][c,t] - sum(y[c,e,t] for e in model.employee_with_competencies[c]))
+                # if(delta[c,t] - abs(model.delta["plus"][c,t].x - model.delta["minus"][c,t].x)) != 0:
+                #     print(delta[c,t] - abs(model.delta["plus"][c,t].x - model.delta["minus"][c,t].x))
+                #     print("Different Delta")
     return delta
 
-
-def calculate_negative_deviation_from_contracted_hours(data, y):
-
-    time_periods_in_week = data["time"]["periods"][2]
+def calculate_negative_deviation_from_contracted_hours(model, y):
     delta_negative_contracted_hours = {}
-
-    for e in data["staff"]["employees"]:
-        for j in data["time"]["weeks"]:
-            delta_negative_contracted_hours[e, j] = data["staff"]["employee_contracted_hours"][
-                e
-            ] - sum(
-                data["time"]["step"] * y[c, e, t]
-                for t in time_periods_in_week[j]
-                for c in data["competencies"]
-            )
+    for e in model.employees:
+        for j in model.weeks:
+            delta_negative_contracted_hours[e,j] = (model.contracted_hours[e]
+            - sum(model.time_step * y[c,e,t] 
+            for c in model.competencies
+            for t in model.time_periods_in_week[c, j]))
     return delta_negative_contracted_hours
 
-
-def calculate_partial_weekends(data, x):
-
+def calculate_partial_weekends(model, x):
     partial_weekend = {}
-
-    for i in data["time"]["saturdays"]:
-        for e in data["staff"]["employees"]:
-
-            partial_weekend[e, i] = abs(
-                sum(x[e, t, v] for t, v in data["shifts"]["shifts_per_day"][i])
-                - sum(x[e, t, v] for t, v in data["shifts"]["shifts_per_day"][i + 1])
-            )
+    partial_weekend_shifts = []
+    for i in model.saturdays:
+        for e in model.employees:
+            #if(abs(sum(model.x[e,t,v] for t,v in model.shifts_at_day[i]) - sum(x[e,t,v] for t,v in model.shifts_at_day[i+1])) != 0):
```

Comment:
Delete?

Reply:
Yes. My merging have included comments I use in my branch that I usually delete.